apps/src/code-studio/components/progress/lessonLockDialog/LessonLockDialog.jsx

```diff
@@ -76,7 +76,19 @@ function LessonLockDialog({
     );
   };
 
+  /*
+  Checks that the user is trying to save new information, otherwise closes
+  the dialog without sending to api post method.
+  */
   const handleSave = async () => {
+    if (serverLockState === clientLockState) {
```

Comment:
I think it would be better to use [lodash's](https://masteringjs.io/tutorials/fundamentals/compare-arrays) `isEqual` here. Array comparison just compares references so if you change something then change it back, you get the "There's nothing to save" error.

[source](https://masteringjs.io/tutorials/fundamentals/compare-arrays)

Reply:
Oh good call- will do!